pkgs/by-name/do/dorion/package.nix

```
@@ -1,65 +1,160 @@
+    inherit version src;
+
+    pnpmDeps = pnpm.fetchDeps {
+      inherit pname version src;
+      hash = "sha256-0++DvoDPY2AnLbZ/PMnxMQL60bqbCsWNvlOan7rTm8k=";
+    };
+
+    nativeBuildInputs = [
+      pnpm.configHook
+      nodejs
+    ];
+
+    buildPhase = ''
+      runHook preBuild
+
+      cp ${shelter} src-tauri/injection/shelter.js
+      pnpm build
+
+      runHook postBuild
+    '';
+
+    installPhase = ''
+      runHook preInstall
+
+      mkdir -p $out
+      cp -r src-tauri/{injection,html} $out
+
+      runHook postInstall
+    '';
+  };
+
+in
+
+rustPlatform.buildRustPackage {
+  inherit pname version src;
+
+  sourceRoot = "${src.name}/src-tauri";
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "rsrpc-0.13.0" = "sha256-2zRuhtVHIbb1yEot2tT49cl0ESfYFvBX5mWu4HtiH7I=";
+      "tauri-plugin-window-state-0.1.1" = "sha256-Mf2/cnKotd751ZcSHfiSLNe2nxBfo4dMBdoCwQhe7yI=";
+      "window-shadows-0.2.2" = "sha256-qSVDA3SYXi40HSYD9VkfBV1hlduC0xp2kAnXGNtH6UE=";
+      "window_titles-0.1.0" = "sha256-lk2T+6curAwqOUuQ8RtYCjX2ygGBgzt4ILBAMV+ql0w=";
+    };
+  };
 
   nativeBuildInputs = [
+    pkg-config
+    yq-go
     autoPatchelfHook
-    dpkg
     wrapGAppsHook3
   ];
 
   buildInputs = [
-    glib-networking
+    openssl
+    webkitgtk'
     gst_all_1.gst-plugins-bad
     gst_all_1.gst-plugins-base
     gst_all_1.gst-plugins-good
-    webkitgtk
+    glib-networking
+  ];
+
+  runtimeDependencies = [
+    libappindicator
+    libayatana-appindicator
   ];
 
-  installPhase = ''
-    runHook preInstall
+  env = {
+    TAURI_RESOURCE_DIR = "${placeholder "out"}/lib";
+  };
 
-    mkdir -pv $out
-    mv -v {bin,lib,share} $out
+  postPatch = ''
+    # disable pre-build script, change distribution dir from an URL to src, and disable auto-updater
+    yq -iPo json '. + {
+      "build": .build + {
+        "beforeBuildCommand": "",
+        "distDir": "src"
+      },
+      "tauri": .tauri + {
+        "bundle": .tauri.bundle + {
+          "resources": [.tauri.bundle.resources[] | select(. | contains("updater") | not)]
+        }
+      }
+    }' tauri.conf.json
+
+    (cd $cargoDepsCopy/tauri-utils-* && patch -p3 <${./tauri-env-resource-dir.patch})
+  '';
 
-    runHook postInstall
+  preConfigure = ''
+    cp -r ${frontend}/* ./
   '';
 
+  postInstall = ''
+    mkdir -p $out/lib/dorion
+    cp -r {injection,html} $out/lib/dorion
+  '';
+
+  passthru = {
+    inherit frontend;
+  };
+
   meta = {
     homepage = "https://github.com/SpikeHD/Dorion";
     description = "Tiny alternative Discord client";
     license = lib.licenses.gpl3Only;
     mainProgram = "dorion";
-    maintainers = [ ];
-    platforms = lib.intersectLists (lib.platforms.linux) (lib.platforms.x86_64);
-    sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
+    maintainers = with lib.maintainers; [ nyanbinary ];
+    platforms = lib.platforms.linux;
+    sourceProvenance = [ lib.sourceTypes.binaryBytecode ];
```

Just to be sure, since this contains both
- readable source code
- bytecode (indeed an unreadable source code, but let's not be so picky)

I suggest

```suggestion
    sourceProvenance = [ 
      lib.sourceTypes.binaryBytecode
      lib.sourceTypes.fromSource 
    ];
```

---

done :3 x2